ArduCopter/mode_throw.cpp

```diff
@@ -272,8 +272,22 @@ bool ModeThrow::throw_detected()
     // Check if the accel length is < 1.0g indicating that any throw action is complete and the copter has been released
     bool no_throw_action = copter.ins.get_accel().length() < 1.0f * GRAVITY_MSS;
 
-    // High velocity or free-fall combined with increasing height indicate a possible air-drop or throw release
-    bool possible_throw_detected = (free_falling || high_speed) && changing_height && no_throw_action;
+    // fetch the altitude above home
+    float altitude_above_home;  // Use altitude above home if it is set, otherwise relative to EKF origin
+    if (ahrs.home_is_set()) {
+        ahrs.get_relative_position_D_home(altitude_above_home);
+        altitude_above_home = -altitude_above_home; // altitude above home is returned as negative
+    }
+    else {
+        altitude_above_home = inertial_nav.get_position_z_up_cm() * 0.01f; // centimeters to meters
+    }
+
+    // Check that the altitude is within user defined limits
+    const bool height_within_params = (altitude_above_home > g.throw_altitude_min) && (g.throw_altitude_max == 0 || (altitude_above_home < g.throw_altitude_max));
```

Comment:
A slight concern is that there's no way to disable the minimum altitude check and the altitude could negative if home has been manually moved higher than the vehicle's takeoff location.   It's an edge case but it could happen.

Reply:
Good point, added a way to disable it same as for the max altitude check.